### DISCUSSION TITLE
Combining Modules and ParallelContainers

### DISCUSSION THREAD

**Discussion by gheorghestrimtu** [testcontainers/testcontainers-go#2274] [Q&A]
I really like the Modules pattern you guys have introduced for encapsulating one Container in a reusable manner. But looking at the examples in the repo I see that all modules use only `testcontainers.GenericContainer` to deploy one container at a time. I would be interested to use Modules and ParallelContainers together. So how would you approach this task?
I see two ways:
1. Encapsulate more than 1 container inside a module, and use `ParallelContainers` inside the module. But this would break the `func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer)` template, because now I have to pass opts for more than one container.
2. Have the module return a `GenericContainerRequest` and use `ParallelContainers` outside the module, but again we would not use `func RunContainer` but another pattern.

WDYT ?

**Comment by mdelapenya** [comment]
Hi @gheorghestrimtu thanks for raising this concern. How would it be your desired way of using this feature of parallel containers plus modules? Parallel containers came before the modules approach, and they are actually not connected, so we are in a good position to design how we want to work with both.
